File: hugh_luscombe/week01/mta-1/js/mta.js
//MTA
const trainLines = {
  'N': ['Times Square','34th','28th','23rd','Union Square','8th'],
  'L': ['8th','6th','Union Square','3rd','1st'],
  '6': ['Grand Central','33rd','28th','23rd','Union Square','Astor Place']
}

//main function
const planTrip = function(line1, station1, line2, station2) {

  // index of start of trip
  let start = trainLines[line1].indexOf(station1);

  // index of trip until Union Square
  let unionSquareFirst = trainLines[line1].indexOf('Union Square');

  // start of trip from Union Square
  let unionSquareSecond = trainLines[line2].indexOf('Union Square');

  // index of end of trip
  let stop = trainLines[line2].indexOf(station2);

  let stations1 = trainLines[line1];
  let stations2 = trainLines[line2];

  //first trip
  let numberOfStopsFirst = 0;
  let stopNames1 = " ";
  if (start < unionSquareFirst) { // look into this section
    for (var i = start+1; i <= unionSquareFirst; i++) {
      numberOfStopsFirst++;
      stopNames1 += stations1[i] + ", ";
    }
  } else {
    for (var i = start-1; i >= unionSquareFirst; i--) {
      numberOfStopsFirst++;
      stopNames1 += stations1[i] + ", ";
    }
  }

  //second trip
  let numberOfStopsSecond = 0;
  let stopNames2 = "";
  if (unionSquareSecond > stop) {
    for (var i = stop; i < unionSquareSecond; i++) {
      numberOfStopsSecond++;
      stopNames2 = stations2[i] + ", " + stopNames2;
    }
  } else {
    for (var i = stop; i > unionSquareSecond; i--) {
      numberOfStopsSecond++;
      stopNames2 = stations2[i] + ", " + stopNames2;
    }
  }

  let stops = numberOfStopsFirst + numberOfStopsSecond;

  console.log(`You must travel through the following stops on the ${line1} line: ${stopNames1}`);
  if (numberOfStopsSecond > 0) {
    console.log(`Change at Union Square. Your journey continues through the following stops on the ${line2} line: ${stopNames2}`);
  }
  console.log(`You have ${stops} stops in total`);
};

planTrip('N', 'Times Square', 'N', 'Union Square');
planTrip('N', 'Times Square', '6', '33rd');
planTrip('6', '33rd', 'L', '8th');
planTrip('L', '1st', 'N', 'Times Square');

File: max_reeves/ruby04/MTAII/mta.rb
# There are 3 subway lines:
# The N line has the following stops: Times Square, 34th, 28th, 23rd, Union Square, and 8th
# The L line has the following stops: 8th, 6th, Union Square, 3rd, and 1st
# The 6 line has the following stops: Grand Central, 33rd, 28th, 23rd, Union Square, and Astor Place.
# All 3 subway lines intersect at Union Square, but there are no other intersection points. (For example, this means the 28th stop on the N line is different than the 28th street stop on the 6 line, so you'll have to differentiate this when you name your stops in the arrays.)
# Tell the user the number of stops AND the stops IN ORDER that they will pass through or change at.
# Hints:

# All lines intercect at Union Square!


n_line = ["Times Square", "34th", "28th.n", "23rd.n", "Union Square", "8th"]
l_line= ["8th", "6th", "Union Square", "3rd", "1st"]
six_line= ["Grand Central", "33rd", "28th.s", "23rd.s", "Union Square", "Astor Place"]

def tripPlanner (first_line, first_stop, last_line, last_stop)
	# if end_stop is on the "firstline", then print all stops from "firststop" to "endstop"
	if  (first_line === last_line)
		first_stop_i = first_line.index(first_stop)
		last_stop_i = first_line.index(last_stop)
		puts first_stop_i
		puts last_stop_i
		puts (first_line[first_stop_i]..first_line[last_stop_i]).to_a
		(first_line[first_stop_i]..first_line[last_stop_i]).to_a.flatten.each do |stop|
			puts stop
		end
	end
end

# 	# else print stops to union square
# 	if  (first_line == n_line && end_stop == "Union Square")
# 		puts n_line[0..4]
# 	end
# 	# then loop through "second_line" array and print stops from union square to "end_stop"
#  	if (first_stop == "Union Square" && second_line == l_line)
# 		l_line[2].each do |numl|
# 			    puts numl
# 	end
# 	if (first_stop == "Union Square" && second_line == six_line)
# 		six_line[4].each do |nums|
# 					puts nums
# 	end
#
# end




# `puts` shows output similar to this:
# "You must travel through the following stops on the N line: 34th, 28th, 23rd, Union Square."


# "Change at Union Square."
# "Your journey continues through the following stops: 23rd, 28th, 33rd."
# "7 stops in total."

tripPlanner(n_line ,"34th", n_line, "8th")
